### PR TITLE
feat(kafkajs.js): add support for kafkajs

### PR DESCRIPTION
**Pull request by ranrib**
Need to update first the epsagon version.

### CODE REVIEW

File: src/wrappers/kafkajs.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            'consume',
+            'trigger'
+        );
+
+        const metadata = {
+            partition: message.partition,
+            offset: message.message.offset,
+            timestamp: new Date(parseInt(message.message.timestamp, 10)).toUTCString(),
+        };
+        if (message.message.headers[EPSAGON_HEADER]) {
```
Comment (galbash) [inline src/wrappers/kafkajs.js]: is this conforming to our java inst?

File: src/wrappers/kafkajs.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        const headers = Object.entries(message.message.headers).reduce((total, entry) => {
+            // eslint-disable-next-line no-param-reassign
+            total[entry[0]] = entry[1].toString();
+            return total;
+        }, {});
+
+        tracer.addEvent(kafkaEvent);
+        eventInterface.finalizeEvent(kafkaEvent, kafkaStartTime, null, metadata, {
+            headers,
+            body: message.message.value.toString(),
```
Comment (galbash) [inline src/wrappers/kafkajs.js]: are you sure this is always string / bytes?

File: src/wrappers/kafkajs.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        if (!options.eachMessage) {
+            return wrappedFunction.apply(this, [options]);
+        }
+        const originalHandler = options.eachMessage;
+        const patchedHandler = message => traceContext.RunInContext(
+            tracer.createTracer,
+            () => kafkaMiddleware(message, originalHandler)
+        );
+        // eslint-disable-next-line no-param-reassign
+        options.eachMessage = patchedHandler;
```
Comment (galbash) [inline src/wrappers/kafkajs.js]: the fact that patchedHandler is an arrow function and you are not using `bind` might lead to trouble. try making a handler that uses `this` and see if this breaks it.

Comment (ranrib) [reply src/wrappers/kafkajs.js]: 👍

File: src/wrappers/kafkajs.js (hunk 2 above)
Comment (ranrib) [reply src/wrappers/kafkajs.js]: Yes:
`The "string" argument must be one of type string, Buffer, or ArrayBuffer`

File: src/wrappers/kafkajs.js (hunk 1 above)
Comment (ranrib) [reply src/wrappers/kafkajs.js]: Fixed

File: src/wrappers/kafkajs.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        }
+
+        if (originalHandler.name) {
+            nodeEvent.getResource().setName(originalHandler.name);
+        }
+
+        // Handle and finalize async user function.
+        if (utils.isPromise(runnerResult)) {
+            let originalHandlerAsyncError;
+            runnerResult.catch((err) => {
```
Comment (galbash) [inline src/wrappers/kafkajs.js]: you need to override runnerResult here

**Pull request merged**